### PR TITLE
feat(core/secrets): decrypt application secrets

### PR DESCRIPTION
**Pull request by KathrynLewis**
Part of secrets management project: [spinnaker/spinnaker#3649](https://github.com/spinnaker/spinnaker/issues/3649)

Adds a Spring PropertySource to allow for decryption of secrets in the services when configs are read in.

### CODE REVIEW

File: kork-core/kork-core.gradle [hunk 1]
```diff
@@ -9,7 +9,7 @@ dependencies {
     compile spinnaker.dependency('spectatorGc')
     compile spinnaker.dependency('spectatorJvm')
     compile spinnaker.dependency('spectatorWebSpring')
-
+    compile project(":kork-secrets-aws")
```
Comment (jkschneider) [inline kork-core/kork-core.gradle]: `kork-core` depends on `kork-secrets-aws`?

Comment (KathrynLewis) [reply kork-core/kork-core.gradle]: It's not required to build kork itself, but it's required when the services try to use the s3 decryption engine.

File: kork-core/src/main/java/com/netflix/spinnaker/kork/secrets/SecretApplicationContextInitializer.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  public void initialize(ConfigurableApplicationContext applicationContext) {
+    applicationContext.getBeanFactory().addBeanPostProcessor(new BeanPostProcessor() {
+      @Override
+      public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+        return bean;
+      }
+
+      @Override
+      public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+        if (beanName.equals("secretManager")) {
```
Comment (robzienert) [inline kork-core/src/main/java/com/netflix/spinnaker/kork/secrets/SecretApplicationContextInitializer.java]: Prefer `"secretManager".equals(beanName)`. I doubt `beanName` is ever going to be null, but I tend to err on the side of absolute caution in this case.

File: kork-core/kork-core.gradle (hunk 1 above)
Comment (robzienert) [reply kork-core/kork-core.gradle]: 🤔 I'm not sure of a better option than what we're doing here, but I'm not enthusiastic about bringing the AWS dependency tree along for the ride into all services even if they're not going to be using the S3 decryption engine.

@ttomsu @cfieber -- Any ideas here? Has this ship already sailed? As more decryption engines come along (GCS, Vault, ...) we're gonna get pretty bloated.

File: kork-core/src/main/java/com/netflix/spinnaker/kork/secrets/SecretApplicationContextInitializer.java (hunk 2 above)
Comment (KathrynLewis) [reply kork-core/src/main/java/com/netflix/spinnaker/kork/secrets/SecretApplicationContextInitializer.java]: fixed

**Comment by KathrynLewis** [comment]
Added tests

File: kork-core/kork-core.gradle (hunk 1 above)
Comment (robzienert) [reply kork-core/kork-core.gradle]: Gonna just say we shipit as is.

**Pull request merged**